### PR TITLE
Comment Sorting

### PR DESCRIPTION
**Pull request by DavidBriglio**

### CODE REVIEW

File: js/cmp/post-comments.js [hunk 1]
```diff
@@ -10,21 +10,37 @@ import LoadMoreComments from './load-more-comments.js'
...
   let loading = true
+
   // load comments
-  getComments(post).then(data => {
-    comments = data
-    loading = false
-    m.redraw()
-  })
+  const loadComments = sort => {
+    getComments(post, null, sort).then(data => {
```
Comment (fuzetsu) [inline js/cmp/post-comments.js]: shouldn't we set `loading = true` when `loadComments` is called to indicate the state on the screen?
```suggestion
    loading = true
    getComments(post, null, sort).then(data => {
```

File: js/util.js [hunk 2]
```diff
@@ -111,4 +111,15 @@ export const reduceCount = (count, digits = 1) => {
 
   // Do not reduce if we are below 1000
   return divisor ? (count / divisor).toFixed(digits) + indicator : count
-}
\ No newline at end of file
+}
+
+export const SortingOptions = [
```
Comment (fuzetsu) [inline js/util.js]: would it make more sense for this to live in `constants.js` ?

File: js/api.js [hunk 3]
```diff
@@ -29,14 +29,15 @@ export const searchPosts = async (query, sort = true) => {
   return results
 }
 
-export const getComments = (post, comment) =>
+export const getComments = (post, comment, sort = 'new') =>
```
Comment (fuzetsu) [inline js/api.js]: I feel like the default here should be the same as reddit's default.
It's called "Best" or "hot" on the site, not sure which one that corresponds to on the API, maybe it's named differently or you just don't pass a `sort` param  :thinking:

File: js/util.js [hunk 4]
```diff
@@ -111,4 +111,15 @@ export const reduceCount = (count, digits = 1) => {
...
+  { value: 'new', text: 'New' },
+  { value: 'top', text: 'Top' },
+  { value: 'old', text: 'Old' },
+  { value: 'controversial', text: 'Controversial' },
+  { value: 'random', text: 'Random' },
+  { value: 'qa', text: 'QA' },
+  { value: 'live', text: 'Live' },
```
Comment (fuzetsu) [inline js/util.js]: Reading on this mode I think we should probably comment it out for now, only applies to certain special threads and is meant for streaming comments, which we don't currently support.. Not sure if all the other ones can actually be used with any thread.

File: js/cmp/post-comments.js (hunk 1 above)
Comment (DavidBriglio) [reply js/cmp/post-comments.js]: Fixed this

File: js/util.js (hunk 2 above)
Comment (DavidBriglio) [reply js/util.js]: Moved into `constants.js`.

File: js/api.js (hunk 3 above)
Comment (DavidBriglio) [reply js/api.js]: I agree, the default is now confidence / best.

File: js/util.js (hunk 4 above)
Comment (DavidBriglio) [reply js/util.js]: Commented out for now.

**Pull request merged**